be/src/io/hdfs_util.cpp

```diff
@@ -41,6 +44,54 @@ Status create_hdfs_fs(const THdfsParams& hdfs_params, const std::string& fs_name
     return Status::OK();
 }
 
+Status create_hdfs_fs_async(const THdfsParams& hdfs_params, const std::string& fs_name,
+                            hdfsFS* fs) {
+    std::promise<std::pair<Status, hdfsFS>> prom;
+    std::future<std::pair<Status, hdfsFS>> fut = prom.get_future();
+
+    std::thread t(
+            [&](std::promise<std::pair<Status, hdfsFS>> prom) {
```

Comment:
why not just call `create_hdfs_fs_sync()` here?

Reply:
> why not just call `create_hdfs_fs_sync()` here?

for returning multi value